tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/jdbc/JdbcTableManager.java

```diff
@@ -199,58 +202,171 @@ public String getDatabaseName()
     private void insertData(TableName tableName, JdbcTableDataSource dataSource)
     {
         Iterator<List<Object>> dataRows = dataSource.getDataRows();
-        PreparedStatement preparedStatement = null;
-        List<List<Object>> batchRows = new ArrayList<>();
+        if (!dataRows.hasNext()) {
+            return;
+        }
+        try {
+            insertData(tableName, dataRows);
+        }
+        catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void insertData(TableName tableName, Iterator<List<Object>> dataRows)
+            throws SQLException
+    {
+        Loader loader = null;
```

Comment:
Move `columnsCount` before loop and use `try-with-resources` since `Loader` is auto-closeable, maybe?